File: Inflearn/basic/02.K번째 수/answer.py
import sys
import math


def solution(idx):
    _, s, e, k = map(int, input().split())
    num = list(map(int, input().split()))
    cut_num = num[s - 1:e]
    cut_num.sort()
    print(f'#{idx} {cut_num[k-1]}')


if __name__ == "__main__":
    sys.stdin = open("data/in1.txt", "rt")
    tcs = int(input())
    for idx in range(1, tcs+1):
        solution(idx)

File: Programmers/lv3/12900_dp/answer.js
function solution(n) {
  a = [0, 1, 2];

  for (let i = 3; i <= n; i++) {
    a.push((a[i - 1] + a[i - 2]) % 1000000007);
  }

  return a[n];
}

console.log(solution(4));
